controllers/application/ingress_virtual_service.go

```diff
@@ -6,154 +6,99 @@ import (
-}
-
-func (r *ApplicationReconciler) defineRedirectVirtualService(ctx context.Context, application *skiperatorv1alpha1.Application) (*networkingv1beta1.VirtualService, error) {
 	virtualService := networkingv1beta1.VirtualService{
 		ObjectMeta: v1.ObjectMeta{
-			Name:      application.Name + "-http-redirect",
+			Name:      application.Name + "-ingress",
 			Namespace: application.Namespace,
 		},
-		Spec: networkingv1beta1api.VirtualService{
-			ExportTo: ExportToNamespaces,
-			Gateways: r.getGatewaysFromApplication(application),
-			Hosts:    []string{"*"},
-			Http: []*networkingv1beta1api.HTTPRoute{
-				{
-					Match: []*networkingv1beta1api.HTTPMatchRequest{
-						{
-							WithoutHeaders: map[string]*networkingv1beta1api.StringMatch{
-								":path": {
-									MatchType: &networkingv1beta1api.StringMatch_Prefix{
-										Prefix: "/.well-known/acme-challenge/",
-									},
+	}
+
+	var err error
+
+	if len(application.Spec.Ingresses) > 0 {
+		_, err = ctrlutil.CreateOrPatch(ctx, r.GetClient(), &virtualService, func() error {
+
+			err := ctrlutil.SetControllerReference(application, &virtualService, r.GetScheme())
+			if err != nil {
+				r.SetControllerError(ctx, application, controllerName, err)
+				return err
+			}
+			virtualService.Spec = networkingv1beta1api.VirtualService{
+				ExportTo: []string{".", "istio-system", "istio-gateways"},
+				Gateways: r.getGatewaysFromApplication(application),
+				Hosts:    application.Spec.Ingresses,
+				Http: []*networkingv1beta1api.HTTPRoute{
+					{
+						Name: "default-app-route",
+						Route: []*networkingv1beta1api.HTTPRouteDestination{
+							{
+								Destination: &networkingv1beta1api.Destination{
+									Host: application.Name,
 								},
 							},
-							Port: 80,
 						},
 					},
-					Redirect: &networkingv1beta1api.HTTPRedirect{
-						Scheme:       "https",
-						RedirectCode: 308,
-					},
 				},
-			},
-		},
-	}
-
-	err := ctrlutil.SetControllerReference(application, &virtualService, r.GetScheme())
-	if err != nil {
-		r.SetControllerError(ctx, application, controllerName, err)
-		return &virtualService, err
-	}
-
-	r.SetLabelsFromApplication(ctx, &virtualService, *application)
-	util.SetCommonAnnotations(&virtualService)
+			}
 
-	return &virtualService, err
-}
-
-func (r *ApplicationReconciler) defineCommonVirtualService(ctx context.Context, application *skiperatorv1alpha1.Application) (*networkingv1beta1.VirtualService, error) {
-	virtualService := networkingv1beta1.VirtualService{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      application.Name + "-ingress",
-			Namespace: application.Namespace,
-		},
-		Spec: networkingv1beta1api.VirtualService{
-			ExportTo: ExportToNamespaces,
-			Gateways: r.getGatewaysFromApplication(application),
-			Hosts:    application.Spec.Ingresses,
-			Http: []*networkingv1beta1api.HTTPRoute{
-				{
-					Route: []*networkingv1beta1api.HTTPRouteDestination{
+			if application.Spec.RedirectToHTTPS {
```

Comment:
Burde ikke default på denne være true?

Reply:
på `RedirectToHTTPS` i API definisjonen, tenkte du?

Reply:
hensikten med defaulten var på en måte å unngå å endre oppførsel til applikasjonen til produktteamene, selv om det virker som en mer sane default, ja.

Reply:
Tenker vi burde ha som mål om at dette bør være på by default så vi kan si vi er secure by default. Det bør vel ikke brekke noe for noen siden vi bruker gyldige sertfikater. Om vi ikke gjør endringen nå tenker jeg det evt. bør bli en ny PR etterpå hvor man kartlegger om det brekker noe og endrer default fra false til true der. Hva tenker du?

Reply:
![image](https://user-images.githubusercontent.com/61270418/231439244-d9954af2-2ef0-4b13-bf9b-171a08b53353.png)

Hadde litt samme prat med Even. Synes vi burde bytte, men vente til vi har sagt i fra om at det er en ting.

Det bør jo helst fungere, men vi kan jo ikke garantere for at produktteam ikke av en eller annen grunn ikke "tillater" https trafikk internt. Det haddde vært rart, men likeså @esphen

Reply:
Da tar vi det opp på plattformlauget onsdag etter kubecon og sier det kommer redirect til https innen 14 dager og ser om noen protesterer 🙂